src/models/state/wallet/address/address_type.rs

```diff
@@ -218,23 +227,66 @@ impl ReceivingAddress {
     pub fn to_bech32m(&self, network: Network) -> Result<String> {
         match self {
             Self::Generation(k) => k.to_bech32m(network),
-            Self::Symmetric(_k) => bail!("bech32m not implemented for symmetric keys"),
+            Self::Symmetric(k) => k.to_bech32m(network),
         }
     }
 
-    /// parses an address from its bech32m encoding
+    /// returns an abbreviated address.
+    ///
+    /// The idea is that this suitable for human recognition purposes
+    ///
+    /// ```text
+    /// format:  <hrp><start>...<end>
+    ///
+    ///   [4 or 6] human readable prefix. 4 for symmetric-key, 6 for generation.
+    ///   8 start of address.
+    ///   8 end of address.
+    /// ```
+    ///
+    /// security: note that if this is used on a symmetric key it will display 16 chars
+    /// of the bech32m encoded key.  This seriously reduces the key's strength and it
+    /// may be possible to brute-force it.  In general it is best practice to avoid
+    /// display of any part of a symmetric key.
```

Comment:
yes, I considered that and perhaps it is the best thing, security wise.

It does introduce an asymmetry between the symmetric display logic and the generation display.   And its not something that should be done only as a hack for dashboard... ie, it should be impl'd for ReceivingAddress.   I will look into a clean way of doing it.  This can have implications for the offchain stuff as well.

Reply:
It's not necessary for ReceivingAddress: leaking characters there does expose secret key information. That's by design ;-)

Reply:
> It's not necessary for ReceivingAddress: leaking characters there does expose secret key information. That's by design ;-)

I think you mean "does NOT expose secret key information".  yes?

Also, I think we may be having a miscommunication due to naming.   When I say `ReceivingAddress` I mean exactly `address_type::ReceivingAddress` which abstracts over both `GenerationReceivingAddress` and `SymmetricKey`.  Likewise when I refer to `SpendingKey` I mean `address_type::SpendingKey`, not `GenerationSpendingKey`.

I believe you are thinking of `GenerationReceivingAddress` instead in the above statement.

Reply:
> I think you mean "does NOT expose secret key information". yes?

Yes. Oops! I edited it to avoid future confusion.

> I believe you are thinking of GenerationReceivingAddress instead in the above statement.

Yes, correct. I guess I never considered the possibility that symmetric receiving addresses could be semantically loaded. In fact, I'm still skeptical, but for now it suffices to confirm your reading of my message.

Reply:
9f55aad01a232d783b5285ad9ecc10f4d4ee88e8 adds ReceivingAddress::to_display_bech32m_abbreviated() which is now used by the dashboard address screen.  For GenerationReceivingAddress it is the same as before, but for SymmetricKey it internally calls ::privacy_preimage() for a hash before bech32m encoding and abbreviating.   @aszepieniec please confirm if you think this is ok/improvement.